### DISCUSSION TITLE
Fee varies according to UTXO age, not my idea

### DISCUSSION THREAD

**Discussion by Shymaa-Arafat** [mit-dci/utreexo#314] [Ideas]
Through this discussion
https://bitcointalk.org/index.php?topic=5354379.new;topicseen#new
Someone suggested this idea
![Screenshot_2021-08-18-08-44-25-84](https://user-images.githubusercontent.com/83260375/129850927-8dbc519c-667f-4f86-998d-fa13cf0a384b.jpg)

I think the idea matches ur Forest design; ie fee is a fn of which power of 2 tree the UTXO in.
.
If u could also discuss the idea of partitioning the UTXOS set; ie separate tree/forest for dust UTXOS (I understand their exact ratio is  ~10%?)
Burned, non-standard,... whatever is not spendable & what is not expected to be spent in the near future; then the regular flow.
.
I'm not the first one to bring that up, u r sure familiar with the paper Jan/2019
https://royalsocietypublishing.org/doi/10.1098/rsos.180817
some words from their conclusion:
"_**Similarly, one can think of a two-tier data structure where a UTXO subset containing UTXOs with a low probability of being selected such as dust is kept on disk, while the other UTXOs are kept in memory."**_

**Comment by adiabat** [comment]
A miner can include whatever transactions they want, even those without any fees at all.  So if the idea is that a miner could prioritize transactions which have small proofs in utreexo, sure, totally possible.

But also irrational.  The miner has a limited resource -- block space -- and is trying to make as much money as possible by getting the fees from filling it.  Picking based on anything other than sat/vByte is probably going to make them lose money.

Unless of course you made a *consensus* change that requires some minimum fee based on utreexo proof.  That seems very unlikely to happen, for a number of reasons.  Utreexo doesn't require any soft or hard forks which is nice.  A consensus requirement would make it hard to change how the accumulator works if better methods are later found.  Also it's not clear that large proofs are something you want to discourage; in the extreme case where transaction fees for a utxo increase rapidly (say they double every day the utxo lasts) everyone would want to move their coins just so the utxos stay "fresh".  Depending on how quickly proof sizes and proof fees ramp up, compared to the normal transaction inclusion fees we have now, you would get some level of "churn" where transactions are made just to refresh utxos and make them newer.  This seems like a bad idea because then you're introducing a large amount of "noise" data into the system and making legitimate usage more expensive.

You could say that's what you want (etherum folks call something like this "state rent"; since your utxo is taking up space on everyone's hard drive, you should pay for that space) but with accumulators, that extra utxo doesn't bother anyone anymore.

**Reply by Shymaa-Arafat** [reply]
What if we partition on BTC value?
Looking at this statistical figure, 
![E-BjHRmVkAQZFQL](https://user-images.githubusercontent.com/83260375/131448933-91b9a3dd-b049-48de-8ee8-dd80dbdf85e3.jpeg)

1-we can find a threshold that almost equally split the UTXOS set.

2-I think there will be more Locality of reference this way, since investors of the same category tend to move their money together according to the same market parameters

-This an older fig (Jan 2021) from
https://insights.glassnode.com/bitcoin-supply-distribution/
I think the ratios doesn't differ so much
![sup-1](https://user-images.githubusercontent.com/83260375/131449622-6ea2b219-c872-4576-891c-8e5ffce3b477.png)

**Reply by adiabat** [reply]
What do you mean by "partition on BTC value"?  Charge higher fees based on the value moved in the transaction?

One, this has nothing to do with utreexo.  Two, it still has the problem that it's irrational for miners, so any miner doing it loses money compared to one that doesn't.

**Reply by Shymaa-Arafat** [reply]
No No, sorry
I mean partitioning the Utreexo tree or forest according to some BTC value that almost half the UTXO set, or to buckets of nearly equal no of UTXOS in each

**Reply by adiabat** [reply]
It's possible to make different trees based on utxo BTC amounts, but my guess is that will lead to larger proofs overall.  UTXO age seems easier because: 
- You get the insertions clustered for free: all utxos of the same age show up at the same time, so insert them all next to each other.
- It's a simple metric: young utxos are more likely to be spent than old utxos.

By contrast, looking at amount size:
- They don't show up at the same time; every block has a wide range of utxo amounts
- I don't know if there's an easy rule like "smaller utxos are more likely to be spent".  

I'm sure you can show that there's a power law distribution of the amounts of utxo spent.  But that will likely be the same distribution as is present in the utxo set itself.  In other words, given a utxo chosen uniformly randomly from the entire utxo set, does it's BTC amount tell me anything about the likelyhood it will be spent in the next block?  Maybe not.  (Though that would be interesting data to have.)
(I guess there are edge cases, like dust is less likely to be spent)

**Reply by Shymaa-Arafat** [reply]
But it would be getting the proof from a tree (or forest) of size N/2 or N/(no of clusters) in general instead of N; leading to a shorter path length
The same metric of younger UTXOS get spent faster could work in the same manner inside each cluster
